### ISSUE TITLE
(feature) add kubernetes config retriever

### ISSUE THREAD

**Issue by marcosnils** [thomaspoignant/go-feature-flag#213]
# Requirements

Would you be interested in contributions to fetch configs from a kubernetes config map? I'd like to give it a try if that's something that you're willing to add here.

**Comment by thomaspoignant** [comment]
@marcosnils sorry I missed this issue before Today.
Definitely, I could be a nice addition to retrieve the flags from a config map.

Fill free to open a pull request for this.

**Comment by thomaspoignant** [comment]
Now I am curious @marcosnils, how do you want to implement that? 
Would you read directly using the kubernetes API?

**Comment by thomaspoignant** [comment]
@marcosnils that is a great idea, even if you can still use your `configmap` as a file and use `go-feature-flag` from there.
I will look at this as soon as I have some time.

**Comment by mschoenlaub** [comment]
@thomaspoignant I could take that over, I guess. I have some free time and would probably enjoy the Go exercise :)

**Comment by thomaspoignant** [comment]
Yes @mschoenlaub if you have time it will be awesome to implement it.

Feel free to ask any questions you have about the implementation and how to do it.
